src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/ExportFileController.cs

```diff
@@ -54,7 +54,7 @@ public async Task<IActionResult> Execute(int id)
 
             using (var fileBuilder = new TemporaryFileBuilder())
             {
-                archiveFileName = PathExtensions.Combine(Path.GetTempPath(), filename);
+                archiveFileName = fileBuilder.Folder + ".zip";
```

Comment:
Uses temporary name for archive (same as temp folder to be compressed) instead  resulting name. Resulting file still named with *filename* as before.